cmd/builder/test/test.sh

```diff
@@ -69,28 +66,29 @@ test_build_config() {
         # Since the content of the servicez page depend on which extensions are
         # built into the collector, we depend only on the zpages extension
         # being present and serving something.
-        curl --fail --silent --output /dev/null http://localhost:55679/debug/servicez
-        if [ $? == 0 ]; then
+        if curl --fail --silent --output /dev/null http://localhost:55679/debug/servicez; then
             echo "✅ PASS ${test}"
 
             kill "${pid}"
-            if [ $? != 0 ]; then
-                echo "Failed to stop the running instance for test ${test}. Return code: $? . Skipping tests."
+            ret=$?
+            if [ "$ret" != 0 ]; then
```

Comment:
Will this compare automatically turn the string `"$ret"` into an int for the comparison?

Reply:
I think they were compared as strings because of the use of operator !=.  I updated the if statement to use numeric comparison and removed the quotes as they were unnessecary.